composer/distributed/param_init.py

```diff
@@ -6,10 +6,15 @@
 from typing import Callable, Optional
 
 import torch.nn as nn
-from torch.distributed.tensor import DTensor
 
+from composer.utils import dist
 
-def meta_init(module: nn.Module, param_init_fn: Optional[Callable[[nn.Module], None]] = None) -> None:
+
+def meta_init(
+    module: nn.Module,
+    param_init_fn: Optional[Callable[[nn.Module], None]] = None,
+    sync_module_states: bool = False,
+) -> None:
```

Comment:
i'm not sure i really understand the logical flow here any more. don't we just want to do something like

```
if not sync_module_states:
    same logic as before calling this meta_init function
else:
    call sync_module_states function, which will call to_empty on all ranks with their current device. the model will have already been sharded, so this should be safe?